src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -706,6 +707,23 @@ private Binder VisitTypeDeclarationCore(TypeDeclarationSyntax parent, NodeUsage
                 return resultBinder;
             }
 
+            private static Symbol ContainingMemberOrLambdaSkippingSimpleProgramEntryPoint(Binder binder)
+            {
+                var containingMemberOrLambda = binder.ContainingMemberOrLambda;
```

Comment:
> From the above, I guess this sometimes (always?) returns types or namespaces. Why is it called ContainingMemberOrLambda?

Because some binders return members or lambdas or even local functions. This PR doesn't change this behavior of the API.

---
In reply to: [382851209](https://github.com/dotnet/roslyn/pull/41706#discussion_r382851209) [](ancestors = 382851209)